homeassistant/components/minecraft_server/api.py

```diff
@@ -82,12 +89,22 @@ async def async_get_data(self) -> MinecraftServerData:
         """Get updated data from the server, supporting both Java and Bedrock Edition servers."""
         status_response: BedrockStatusResponse | JavaStatusResponse
 
-        try:
-            status_response = await self._server.async_status()
-        except OSError as error:
-            raise MinecraftServerConnectionError(
-                f"Fetching data from the server failed (error: {error})"
-            ) from error
+        for trial in range(DATA_UPDATE_RETRIES):
+            try:
+                status_response = await self._server.async_status(tries=1)
```

Comment:
I have a couple questions:
- Why do tries here instead of within the library? (there is a tries arugment ...)
- Where is `tries` consumed in this library? I don't see it as an argument to `AsyncServerPinger` but maybe i am looking in the wrong plce
- The functions appear to already have retry decorators that have 3 tries:
https://github.com/py-mine/mcstatus/blob/711ea586c2040ce35c3f4b837d1f4061b040c4c5/mcstatus/server.py#L147 -- is this intentionally to do 3^3 retries? Seems a bit much

Reply:
The default of the library `mcstatus` is 3 trials in total (if the `tries` argument is not used). As I wrote in the PR description, I found out during testing, that sometimes the library does not retry properly, for example if an exception occurs during fetching of updated data. After looking into the the `retry` decorator, I think I know the reason: If a retry is expected after some exception is raised, then this exception also has to be added to the retry decorator ([link](https://pypi.org/project/retry/)), but in the library it is not the case. Hence the own retry handling in this integration.

Regarding 3^3 tries: In the retry handling of this integration, the `async_status `API is called with `tries=1` -> No retry from the library.

Reply:
One of my questions though is where is `tries` consumed exactly? Are you saying it is consumed by the @retry decorator? When I try that I don't see it working.

```
>>> import retry
>>> @retry.retry(tries=3)
... def myfunc(**kwargs):
...   print("Attempt")
...   raise ValueError("fail")
... 
>>> myfunc()
Attempt
Attempt
Attempt
Traceback (most recent call last):
...
ValueError: fail
>>> myfunc(tries=5)
Attempt
Attempt
Attempt
Traceback (most recent call last):
ValueError: fail
```

I'm not sure I agree about your assessment of how exceptions are caught.  The retry decorator catches all Exceptions by default.


Perhaps we can back up a bit here: What's failing, exactly, that we're trying to be more resilient to?

Reply:
Thank you so much for testing it. I was very sure that the retry mechanism of the library was not working in case of exceptions, but with your test I am not anymore. I'll look into it in more detail. And if it turns out to be working, I'll remove the retry handling in my integration and just increase the timeouts 👍

Reply:
Okay, after some research and debugging I found out following information:
- The library `mcstatus` doesn't use the library `retry`, instead it uses its [own retry decorator](https://github.com/py-mine/mcstatus/blob/9e46b90afe5782f359925ca6cd53364394391681/mcstatus/utils.py#L24-L75)
- But it retries even in case of exceptions, just like your tests with the default `retry`. And if all retries failed, it will raise the last exception

Therefore I'll remove my retry handling, sorry for the inconvenience and thanks for your support 👍